fastmri_recon/data/datasets/multicoil/preprocessing.py

```diff
@@ -1,8 +1,11 @@
 import tensorflow as tf
+from tfkbnufft import kbnufft_forward, kbnufft_adjoint
+from tfkbnufft.mri.dcomp_calc import calculate_radial_dcomp_tf
 
 from ...utils.masking.gen_mask_tf import gen_mask_tf
-from ...utils.multicoil.smap_extract import extract_smaps
-from ....models.utils.fourier import tf_unmasked_adj_op
+from ...utils.multicoil.smap_extract import extract_smaps, non_cartesian_extract_smaps
+from ....models.utils.fourier import tf_unmasked_adj_op, tf_unmasked_adj_op_non_cartesian, nufft
+from ...utils.non_cartesian import get_radial_trajectory, get_debugging_cartesian_trajectory
```

Comment:
can you add spiral too ?

Reply:
Wait, is spiral initialization up? 
How good is the density comnpensator? 
From what i remember, you density compensation was in a way very specific to radial trajectories and this could be worked on also.... Although it is an issue in tfkbnufft. 
I will see more in detail and open an issue there.

Reply:
Yes yes I am using spiral (not as an intialization though just as a trajectory).
Well the density compensation seems to work fine at least in the network, I didn't try it on its own, maybe I could do that.

It might be that my density compensation implementation is specific to radial trajectories, I didn't look too much into it.